docker/Dockerfile

```diff
@@ -0,0 +1,188 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+# Description:
+#   Builds the environment needed to build Avalon shell.
+#
+#  Configuration (build) parameters
+#  - proxy configuration: https_proxy http_proxy ftp_proxy
+#
+# Build:
+#   $ docker build docker -f docker/Dockerfile -t avalon-shell-dev
+#   if behind a proxy, you might want to add also below options
+#   --build-arg https_proxy=$https_proxy --build-arg http_proxy=$http_proxy --build-arg ftp_proxy=$ftp_proxy
+
+# -------------=== build avalon shell image ===-------------
+FROM ubuntu:bionic as base_image
+
+# Ignore timezone prompt in apt
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Add necessary packages
+RUN apt-get update \
+ && apt-get install -y -q \
+    software-properties-common \
+    python3-pip \
+    python3-toml \
+    python3-requests \
+    python3-colorlog \
+    python3-twisted \
+ && apt-get -y -q upgrade \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+# Make Python3 default
+RUN ln -sf /usr/bin/python3 /usr/bin/python
+
+# Installing solidity compiler
+RUN add-apt-repository ppa:ethereum/ethereum \
+   && apt-get update \
+   && apt-get install solc
+
+# Install py-solc-x and web3 packages using pip because
+# these are not available in apt repository.
+# Install nose2 for running tests.
+RUN pip3 install --upgrade json-rpc web3 py-solc-x nose2
+
+# Install specific version of solc to support versions <0.6.0
+RUN python3 -m solcx.install v0.5.15
+
+# -------------=== Build openssl_image ===-------------
+
+#Build openssl intermediate docker image
+FROM ubuntu:bionic as openssl_image
+
+RUN apt-get update \
+ && apt-get install -y -q \
+    ca-certificates \
+    pkg-config \
+    make \
+    wget \
+    tar \
+ && apt-get -y -q upgrade \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /tmp
+
+# Build ("Untrusted") OpenSSL
+RUN OPENSSL_VER=1.1.1d \
+ && wget https://www.openssl.org/source/openssl-$OPENSSL_VER.tar.gz \
+ && tar -zxf openssl-$OPENSSL_VER.tar.gz \
+ && cd openssl-$OPENSSL_VER/ \
+ && ./config \
+ && THREADS=8 \
+ && make -j$THREADS \
+ && make test \
+ && make install -j$THREADS
+
+#Build Avalon Enclave Manager intermediate docker image
+FROM base_image as build_image
+
+RUN apt-get update \
+ && apt-get install -y -q \
+    build-essential \
+    pkg-config \
+    cmake \
+    make \
+    git \
+    libprotobuf-dev \
+    libssl-dev \
+    python3-dev \
+    swig \
+    tar
+
+RUN pip3 install --upgrade setuptools
+
+# Copy openssl build artifacts from openssl_image
+ENV OPENSSL_VER=1.1.1d
```

Comment:
Remove the above command. Its not needed.

Reply:
Done

Reply:
Done